### DISCUSSION TITLE
How to make a blacklist at your proxy server

### DISCUSSION THREAD

**Discussion by aleksartamonov** [OutlineFoundation/outline-sdk#258] [Q&A]
I want to forbid to go to some of sites while using my vpn proxy. I added dnsmasq-s for domains . But seems it doesn't work.

**Comment by fortuna** [comment]
You can write a [StreamDialer](https://pkg.go.dev/github.com/Jigsaw-Code/outline-sdk/transport#StreamDialer) or [PacketDialer](https://pkg.go.dev/github.com/Jigsaw-Code/outline-sdk/transport#PacketDialer) that fails for blocked websites.

Or use a DNS resolver that blocks the sites you want (e.g. [NextDNS](https://nextdns.io/)). You can use the [HappyEyeballsStreamDialer](https://pkg.go.dev/github.com/Jigsaw-Code/outline-sdk/transport#HappyEyeballsStreamDialer) with your [resolver](https://pkg.go.dev/github.com/Jigsaw-Code/outline-sdk/dns) of choice.
